extensions/picocli/runtime/src/main/java/io/quarkus/picocli/runtime/PicocliBeansFactory.java

```diff
@@ -0,0 +1,19 @@
+package io.quarkus.picocli.runtime;
+
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.CDI;
+
+import picocli.CommandLine;
+
+class PicocliBeansFactory implements CommandLine.IFactory {
+    private final CommandLine.IFactory defaultFactory = CommandLine.defaultFactory();
+
+    @Override
+    public <K> K create(Class<K> aClass) throws Exception {
+        Instance<K> instance = CDI.current().select(aClass);
```

Comment:
I would urge to use `Arc.container()` here instead of `CDI.current()`

Reply:
Why use `Arc.container()` instead of `CDI.current()`? Isn't `CDI` the standard?

Reply:
Yes, `CDI.current()` is the standard, but this is a Quarkus piece of code so let's go with the slightly more performant option that is also used heavily throughout the codebase

Reply:
I see, thanks for the clarification!

Reply:
You are welcome!